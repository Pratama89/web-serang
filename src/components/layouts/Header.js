"use client";

import { signOut, useSession } from "next-auth/react";
import Image from "next/image";
import Link from "next/link";
import { useEffect, useState } from "react";

export default function Header() {
  const [navbar, setNavbar] = useState(false);
  const [color, setColor] = useState(false);

  // change color sat di scroll
  const changeColor = () => {
    if (window.scrollY >= 90) {
      setColor(true);
    } else {
      setColor(false);
    }
  };

  if (typeof window !== "undefined") {
    window.addEventListener("scroll", changeColor);
  }

  return (
    <nav
      className={
        color
          ? "header-bg dark:bg-gray-900 fixed top-0 left-0 right-0 z-10 transition duration-300 ease-in-out"
          : "bg-transparent  dark:bg-gray-900 fixed top-0 left-0 right-0 z-10 transition duration-300 ease-in-out"
      }>
      <div className="w-full flex flex-wrap items-center justify-between mx-auto px-4 lg:max-w-7xl md:items-center md:flex md:px-8 ">
        <div className="  flex items-center justify-between py-3  md:py-5 sm:w-full xs:w-full mb:block md:w-1/2">
          {/* Logo */}
          <Link href={"/"}>
            <Image
              src="/logolaminating.png"
              alt="Laminating Logo"
              width={300}
              height={300}
              priority
              className="w-64 h-full "
            />
          </Link>
          {/* Hamburger Menu */}
          <div className="md:hidden ">
            <button
              onClick={() => setNavbar(!navbar)}
              type="button"
              className="inline-flex items-center p-2 w-10 h-10 justify-center  text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600">
              {navbar ? (
                <Image src={"/close.svg"} alt="close" width={30} height={30} />
              ) : (
                <Image
                  src={"/hamburger-menu.svg"}
                  alt="menu"
                  width={30}
                  height={30}
                />
              )}
            </button>
          </div>
        </div>
        {/* Menu Items */}
        <div
          className={`flex-1 justify-self-center pb-3 mt-8 md:block md:pb-0 md:mt-0  md:w-1/2 ${
            navbar ? "block p-12 md:p-0" : "hidden"
          }`}
          id="navbar-default">
          <ul className="h-70 md:h-auto items-center justify-center md:flex lg:justify-end  ">
          <li className="pb-6 text-xl text-white py-2 px-4 text-center border-b-2 md:border-b-0 bg-[rgba(0,0,0,0.4)] hover:text-red-600 hover:font-semibold  rounded-lg md:bg-[rgba(0,0,0,0.4)] md:py-2 md:px-4 dark:text-white">
            <Link href="#home" onClick={() => setNavbar(!navbar)} aria-current="page">
              Home
            </Link>
          </li>
            <li className="pb-6 text-xl text-white py-2 px-4 text-center border-b-2 md:border-b-0 bg-[rgba(0,0,0,0.4)] hover:text-red-600 hover:font-semibold   rounded-lg md:bg-[rgba(0,0,0,0.4)] md:py-2 md:px-4 dark:text-white">
              <Link href="#about" onClick={() => setNavbar(!navbar)}>
                About
              </Link>
            </li>
            <li className=" pb-6 text-xl text-white py-2 md:px-6 text-center border-b-2 md:border-b-0 bg-primary hover:text-red-600 hover:font-semibold rounded md:bg-transparent md:text-primary md:p-0 dark:text-white md:dark:text-red-500">
              <Link href="#product" onClick={() => setNavbar(!navbar)}>
                Product
              </Link>
            </li>
            <li className=" pb-6 text-xl text-white py-2 px-6 text-center border-b-2 md:border-b-0 bg-primary hover:text-red-600 hover:font-semibold rounded md:bg-transparent md:text-primary md:p-0 dark:text-white md:dark:text-red-500">
              <Link href="#contact" onClick={() => setNavbar(!navbar)}>
                Contact
              </Link>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
}
