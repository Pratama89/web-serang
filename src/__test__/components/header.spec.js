import Header from "@/components/layouts/Header";
import { render, screen } from "@testing-library/react";
import { describe } from "node:test";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/header",
      pathname: "",
      query: "",
      asPath: "",
      push: jest.fn(),
      events: {
        on: jest.fn(),
        off: jest.fn(),
      },
      beforePopState: jest.fn(() => null),
      prefetch: jest.fn(() => null),
      isReady: true,
    };
  },
}))

describe("Header", () => {
  it("should render", () => {
    const page = render(<Header />);
    // expect(screen.getByTestId("title").textContent).toBe("Header");
    expect(page).toMatchSnapshot();
  });
});

