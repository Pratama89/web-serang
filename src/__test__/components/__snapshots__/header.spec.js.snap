// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="bg-white  dark:bg-gray-900 fixed top-0 left-0 right-0 z-10"
      >
        <div
          class="w-full flex flex-wrap items-center justify-between mx-auto px-4 lg:max-w-7xl md:items-center md:flex md:px-8 "
        >
          <div
            class="  flex items-center justify-between py-3  md:py-5 sm:w-full xs:w-full mb:block md:w-1/2"
          >
            <a
              href="/"
            >
              <img
                alt="Laminating Logo"
                class="w-64 h-full "
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="300"
                src="/_next/image?url=%2Flogolaminating.png&w=640&q=75"
                srcset="/_next/image?url=%2Flogolaminating.png&w=384&q=75 1x, /_next/image?url=%2Flogolaminating.png&w=640&q=75 2x"
                style="color: transparent;"
                width="300"
              />
            </a>
            <div
              class="md:hidden "
            >
              <button
                class="inline-flex items-center p-2 w-10 h-10 justify-center  text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600"
                type="button"
              >
                <img
                  alt="menu"
                  data-nimg="1"
                  decoding="async"
                  height="30"
                  loading="lazy"
                  src="/hamburger-menu.svg"
                  style="color: transparent;"
                  width="30"
                />
              </button>
            </div>
          </div>
          <div
            class="flex-1 justify-self-center pb-3 mt-8 md:block md:pb-0 md:mt-0  md:w-1/2 hidden"
            id="navbar-default"
          >
            <ul
              class="h-70 md:h-auto items-center justify-center md:flex lg:justify-end  "
            >
              <li
                class="pb-6 text-xl text-white py-2 md:px-6 text-center border-b-2 md:border-b-0 bg-primary hover:text-red-600 hover:font-semibold rounded md:bg-transparent md:text-primary md:p-0 dark:text-white md:dark:text-red-500"
              >
                <a
                  aria-current="page"
                  href="#home"
                >
                  Home
                </a>
              </li>
              <li
                class="pb-6 text-xl text-white py-2 md:px-6 text-center border-b-2 md:border-b-0 bg-primary hover:text-red-600 hover:font-semibold rounded md:bg-transparent md:text-primary md:p-0 dark:text-white md:dark:text-red-500"
              >
                <a
                  href="#about"
                >
                  About
                </a>
              </li>
              <li
                class="pb-6 text-xl text-white py-2 md:px-6 text-center border-b-2 md:border-b-0 bg-primary hover:text-red-600 hover:font-semibold rounded md:bg-transparent md:text-primary md:p-0 dark:text-white md:dark:text-red-500"
              >
                <a
                  href="#product"
                >
                  Product
                </a>
              </li>
              <li
                class="pb-6 text-xl text-white py-2 px-6 text-center border-b-2 md:border-b-0 bg-primary hover:text-red-600 hover:font-semibold rounded md:bg-transparent md:text-primary md:p-0 dark:text-white md:dark:text-red-500"
              >
                <a
                  href="#contact"
                >
                  Contact
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="bg-white  dark:bg-gray-900 fixed top-0 left-0 right-0 z-10"
    >
      <div
        class="w-full flex flex-wrap items-center justify-between mx-auto px-4 lg:max-w-7xl md:items-center md:flex md:px-8 "
      >
        <div
          class="  flex items-center justify-between py-3  md:py-5 sm:w-full xs:w-full mb:block md:w-1/2"
        >
          <a
            href="/"
          >
            <img
              alt="Laminating Logo"
              class="w-64 h-full "
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="300"
              src="/_next/image?url=%2Flogolaminating.png&w=640&q=75"
              srcset="/_next/image?url=%2Flogolaminating.png&w=384&q=75 1x, /_next/image?url=%2Flogolaminating.png&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
          </a>
          <div
            class="md:hidden "
          >
            <button
              class="inline-flex items-center p-2 w-10 h-10 justify-center  text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600"
              type="button"
            >
              <img
                alt="menu"
                data-nimg="1"
                decoding="async"
                height="30"
                loading="lazy"
                src="/hamburger-menu.svg"
                style="color: transparent;"
                width="30"
              />
            </button>
          </div>
        </div>
        <div
          class="flex-1 justify-self-center pb-3 mt-8 md:block md:pb-0 md:mt-0  md:w-1/2 hidden"
          id="navbar-default"
        >
          <ul
            class="h-70 md:h-auto items-center justify-center md:flex lg:justify-end  "
          >
            <li
              class="pb-6 text-xl text-white py-2 md:px-6 text-center border-b-2 md:border-b-0 bg-primary hover:text-red-600 hover:font-semibold rounded md:bg-transparent md:text-primary md:p-0 dark:text-white md:dark:text-red-500"
            >
              <a
                aria-current="page"
                href="#home"
              >
                Home
              </a>
            </li>
            <li
              class="pb-6 text-xl text-white py-2 md:px-6 text-center border-b-2 md:border-b-0 bg-primary hover:text-red-600 hover:font-semibold rounded md:bg-transparent md:text-primary md:p-0 dark:text-white md:dark:text-red-500"
            >
              <a
                href="#about"
              >
                About
              </a>
            </li>
            <li
              class="pb-6 text-xl text-white py-2 md:px-6 text-center border-b-2 md:border-b-0 bg-primary hover:text-red-600 hover:font-semibold rounded md:bg-transparent md:text-primary md:p-0 dark:text-white md:dark:text-red-500"
            >
              <a
                href="#product"
              >
                Product
              </a>
            </li>
            <li
              class="pb-6 text-xl text-white py-2 px-6 text-center border-b-2 md:border-b-0 bg-primary hover:text-red-600 hover:font-semibold rounded md:bg-transparent md:text-primary md:p-0 dark:text-white md:dark:text-red-500"
            >
              <a
                href="#contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
